Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/certificate/DccQrCodeExtractor.kt

```diff
@@ -183,7 +186,8 @@ class DccQrCodeExtractor @Inject constructor(
             header = header,
             certificate = body.parsed.asCertificate,
             certificateJson = body.raw,
-            kid = message.kid
+            kid = message.kid,
+            signature = this
```

Comment:
@chiljamgossow  my question is basically, is that right here ` signature = this` ? or it should be different value than `this` ?

Reply:
You need to add this to DccCoseDecoder: 
`private fun CBORObject.extractSignature(): ByteArray = try {
        this[3].GetByteString()
    } catch (e: Exception) {
        throw InvalidHealthCertificateException(HC_COSE_NO_SIGN1)
    }`

Reply:
And extend the Message data class with the signature

Reply:
And add the error code to the exception